### PR TITLE
refactor: Update package_info_plus

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue

Closes: #821

## Approach
Update package_info_plus to ```package_info_plus: ^4.0.2```

## Tasks

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

**Review by mtrezza — CHANGES_REQUESTED** [review]
CI tests failing...

**Comment by mtrezza** [comment]
Re-running CI to get all tests to pass

**Comment by mbfakourii** [comment]
> It seems that the CI is always failing, and it's not just the coverage upload. Could you take a look?

Yes, Sure

**Comment by mbfakourii** [comment]
The last error related to the [code format](https://github.com/parse-community/Parse-SDK-Flutter/actions/runs/5222554583/jobs/9428184149) is not a problem in my system, but it is a problem in CI!

image

**Comment by mtrezza** [comment]
I think the issue is that the linter is part of the dart framework. So if the Dart beta version enforces a new lint rule, the lint check for that framework version will fail. That's why it's only failing for "Dart beta".

**Comment by mbfakourii** [comment]
@mtrezza 
Why should the beta version be checked?

**Comment by mtrezza** [comment]
It fails `dart format`, which I assume is because the linter changed in that dart version.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -33,9 +33,6 @@ jobs:
           - name: Dart 2.18
             os: ubuntu-latest
             sdk: 2.18.7
-          - name: Dart beta
```
Comment (mtrezza) [inline .github/workflows/ci.yml:36]: Why removed?

**Comment by mbfakourii** [comment]
Beta removed from ci

**Comment by mtrezza** [comment]
Why removed?

**Comment by mbfakourii** [comment]
> Why removed? Adding beta was debated thoroughly in a previous PR and we deiced to add it. If we now remove it we should find arguments that outweigh the other arguments.
> 
> I think we'd need to:
> 
> 1. Find out what exactly the linter is complaining about.
> 2. If we can't fix it, remove linting for beta, instead of removing beta altogether.
> 3. If we can't removing linting for beta, decide whether we should remove beta.

I was testing and returned it

**Comment by mtrezza** [comment]
Do you think you could run `dart format` locally with a beta version to find out what the 1 issue is the linter corrects?

**Comment by mbfakourii** [comment]
> Do you think you could run `dart format` locally with the dart beta version to find out what the 1 issue is the linter corrects?

I will test

**Comment by mbfakourii** [comment]
@mtrezza 
It has problems from version 2.18 to 3. I will roll back to the previous version. I think it is better if there is a problem in the beta!

**Comment by mtrezza** [comment]
Is there no way we can slightly refactor the code so it passes both?

**Comment by mbfakourii** [comment]
> Is there no way we can slightly refactor the code so it passes both?

I will check

**Comment by mbfakourii** [comment]
fixed

**Review by mtrezza — APPROVED** [review]
Well done! So we can keep the beat versions in the CI; is this ready for merge?

**Comment by mbfakourii** [comment]
> Well done! So we can keep the beat versions in the CI; is this ready for merge?

Yes, it is ready to merge

**Pull request merged**